examples/wordpress/README.md

```diff
@@ -0,0 +1,160 @@
+```
+gcloud compute instances list | cut -d ' ' -f 1 | tail -n +2 | xargs -n1 gcloud compute ssh --command="sudo apt-get update;sudo apt-get -y install nfs-common"
+```
+
+
+## Understanding the Wordpress example template
+
+Let's take a closer look at the template used by the Wordpress example. The Wordpress application consists of 4 microservices: an nginx service, a wordpress-php service, a MySQL service, and an NFS service. The architecture looks as follows:
+
+![Architecture](architecture.png)
+
+### Variables
+The template contains the following variables:
+
+```
+{% set PROPERTIES = properties or {} %}
+{% set PROJECT = PROPERTIES['project'] or 'dm-k8s-testing' %}
+{% set NFS_SERVER = PROPERTIES['nfs-server'] or {} %}
+{% set NFS_SERVER_IP = NFS_SERVER['ip'] or '10.0.253.247' %}
+{% set NFS_SERVER_PORT = NFS_SERVER['port'] or 2049 %}
+{% set NFS_SERVER_DISK = NFS_SERVER['disk'] or 'nfs-disk' %}
+{% set NFS_SERVER_DISK_FSTYPE = NFS_SERVER['fstype'] or 'ext4' %}
+{% set NGINX = PROPERTIES['nginx'] or {} %}
+{% set NGINX_PORT = 80 %}
+{% set NGINX_REPLICAS = NGINX['replicas'] or 2 %}
+{% set WORDPRESS_PHP = PROPERTIES['wordpress-php'] or {} %}
+{% set WORDPRESS_PHP_REPLICAS = WORDPRESS_PHP['replicas'] or 2 %}
+{% set WORDPRESS_PHP_PORT = WORDPRESS_PHP['port'] or 9000 %}
+{% set MYSQL = PROPERTIES['mysql'] or {} %}                                                                                                                                                                                                                                 {% set MYSQL_PORT = MYSQL['port'] or 3306 %}                                                                                                                                                                                                                                {% set MYSQL_PASSWORD = MYSQL['password'] or 'mysql-password' %}                                                                                                                                                                                                            {% set MYSQL_DISK = MYSQL['disk'] or 'mysql-disk' %}                                                                                                                                                                                                                        {% set MYSQL_DISK_FSTYPE = MYSQL['fstype'] or 'ext4' %}
+```
+
+### Nginx service
+The nginx service is a replicated service with 2 replicas:
+
+```
+- name: nginx
+  type: https://raw.githubusercontent.com/kubernetes/deployment-manager/master/types/replicatedservice/v2/replicatedservice.py
+  properties:
+    service_port: {{ NGINX_PORT }}
+    container_port: {{ NGINX_PORT }}
+    replicas: {{ NGINX_REPLICAS }}
+    external_service: true
+    image: gcr.io/{{ PROJECT }}/nginx:latest
+    volumes:
+      - mount_path: /var/www/html
+        nfs:
+          server: {{ NFS_SERVER_IP }}
```

Comment:
Rather than duplicate the `NFS_SERVER_IP` twice, create a single `PersistentVolume` object for the NFS server and just reuse a named `ReadWriteMany` `PersistentVolumeClaim` object. This makes the `NFS_SERVER_IP` templated in exactly one place and reduces the grossness (see the example I dropped in https://github.com/kubernetes/kubernetes/tree/master/examples/nfs ).